pkg/vsphere/extraconfig/encode.go

```diff
@@ -252,6 +253,9 @@ func encodeWithPrefix(src interface{}, prefix string) []types.BaseOptionValue {
 				switch field.Interface().(type) {
 				case time.Time:
 					config = append(config, &types.OptionValue{Key: key, Value: field.Interface().(time.Time).String()})
+				case net.IPNet:
+					n := field.Interface().(net.IPNet)
+					config = append(config, &types.OptionValue{Key: key, Value: n.String()})
```

Comment:
Ah, yeah you're right!  Even `encoding/json` has issues with this!  Using the Marshaller/Unmarshaller interfaces, I had to do make similar routines.  (https://enatai-git.eng.vmware.com/enatai/enatai/blob/master/enatai-provisioning-adapters/bmp-adapters/src/main/go/src/slingshot/types/ipnet.go)

Ok.  Sounds good.  Thanks for clarifying!